include/libdnf/base/goal.hpp

```diff
@@ -20,14 +20,24 @@ along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
 #ifndef LIBDNF_BASE_GOAL_HPP
 #define LIBDNF_BASE_GOAL_HPP
 
-#include "base.hpp"
 
 #include "libdnf/base/goal_elements.hpp"
 #include "libdnf/base/transaction.hpp"
-#include "libdnf/rpm/package.hpp"
```

Comment:
What is the reason for removing the `package.hpp` include and forward-declaring the `Package`? Visualizing the include tree I think you should be able to include this package header and use the proper `Package` declaration in a high-level header like `goal.hpp`. IMO we should include the headers in these cases even if a forward declaration would be sufficient.

Reply:
I had to do the forward declaration because there were circular imports causing compilation errors.
This seemed to be the easiest fix.

Reply:
It turns out the change is no longer needed. I probably managed to fix the circular import problem while refactoring the other code. Dropping the commit completely.